Mathlib/Data/IsROrC/Basic.lean

```diff
@@ -143,236 +138,208 @@ theorem inv_def (z : K) : z⁻¹ = conj z * ((‖z‖ ^ 2)⁻¹ : ℝ) :=
   re.map_zero
 #align is_R_or_C.zero_re' IsROrC.zero_re'
 
 @[norm_cast]
-theorem of_real_one : ((1 : ℝ) : K) = 1 := by rw [of_real_alg, one_smul]
-#align is_R_or_C.of_real_one IsROrC.of_real_one
+theorem ofReal_one : ((1 : ℝ) : K) = 1 := map_one _
+#align is_R_or_C.of_real_one IsROrC.ofReal_one
 
-@[simp, is_R_or_C_simps]
-theorem one_re : re (1 : K) = 1 := by rw [← of_real_one, of_real_re]
+@[simp, isROrC_simps]
+theorem one_re : re (1 : K) = 1 := by rw [← ofReal_one, ofReal_re]
 #align is_R_or_C.one_re IsROrC.one_re
 
-@[simp, is_R_or_C_simps]
-theorem one_im : im (1 : K) = 0 := by rw [← of_real_one, of_real_im]
+@[simp, isROrC_simps]
+theorem one_im : im (1 : K) = 0 := by rw [← ofReal_one, ofReal_im]
 #align is_R_or_C.one_im IsROrC.one_im
 
 @[norm_cast]
-theorem of_real_inj {z w : ℝ} : (z : K) = (w : K) ↔ z = w :=
-  { mp := fun h => by convert congr_arg re h <;> simp only [of_real_re]
-    mpr := fun h => by rw [h] }
-#align is_R_or_C.of_real_inj IsROrC.of_real_inj
-
-@[simp, is_R_or_C_simps]
-theorem bit0_re (z : K) : re (bit0 z) = bit0 (re z) := by simp only [bit0, map_add]
-#align is_R_or_C.bit0_re IsROrC.bit0_re
-
-@[simp, is_R_or_C_simps]
-theorem bit1_re (z : K) : re (bit1 z) = bit1 (re z) := by
-  simp only [bit1, AddMonoidHom.map_add, bit0_re, add_right_inj, one_re]
-#align is_R_or_C.bit1_re IsROrC.bit1_re
-
-@[simp, is_R_or_C_simps]
-theorem bit0_im (z : K) : im (bit0 z) = bit0 (im z) := by simp only [bit0, map_add]
-#align is_R_or_C.bit0_im IsROrC.bit0_im
-
-@[simp, is_R_or_C_simps]
-theorem bit1_im (z : K) : im (bit1 z) = bit0 (im z) := by
-  simp only [bit1, add_right_eq_self, AddMonoidHom.map_add, bit0_im, one_im]
-#align is_R_or_C.bit1_im IsROrC.bit1_im
-
-theorem of_real_eq_zero {z : ℝ} : (z : K) = 0 ↔ z = 0 := by
-  rw [← of_real_zero] <;> exact of_real_inj
-#align is_R_or_C.of_real_eq_zero IsROrC.of_real_eq_zero
-
-theorem of_real_ne_zero {z : ℝ} : (z : K) ≠ 0 ↔ z ≠ 0 :=
-  of_real_eq_zero.Not
-#align is_R_or_C.of_real_ne_zero IsROrC.of_real_ne_zero
-
-@[simp, is_R_or_C_simps, norm_cast]
-theorem of_real_add ⦃r s : ℝ⦄ : ((r + s : ℝ) : K) = r + s := by
-  apply (@IsROrC.ext_iff K _ ((r + s : ℝ) : K) (r + s)).mpr
-  simp
-#align is_R_or_C.of_real_add IsROrC.of_real_add
-
-@[simp, is_R_or_C_simps, norm_cast]
-theorem of_real_bit0 (r : ℝ) : ((bit0 r : ℝ) : K) = bit0 (r : K) :=
-  ext_iff.2 <| by simp [bit0]
-#align is_R_or_C.of_real_bit0 IsROrC.of_real_bit0
-
-@[simp, is_R_or_C_simps, norm_cast]
-theorem of_real_bit1 (r : ℝ) : ((bit1 r : ℝ) : K) = bit1 (r : K) :=
-  ext_iff.2 <| by simp [bit1]
-#align is_R_or_C.of_real_bit1 IsROrC.of_real_bit1
-
-@[simp, norm_cast, is_R_or_C_simps]
-theorem of_real_neg (r : ℝ) : ((-r : ℝ) : K) = -r :=
-  ext_iff.2 <| by simp
-#align is_R_or_C.of_real_neg IsROrC.of_real_neg
-
-@[simp, norm_cast, is_R_or_C_simps]
-theorem of_real_mul (r s : ℝ) : ((r * s : ℝ) : K) = r * s :=
-  ext_iff.2 <| by simp [is_R_or_C_simps]
-#align is_R_or_C.of_real_mul IsROrC.of_real_mul
```

Comment:
Are we still waiting for one more round of mathport output here?

Reply:
Last time I checked (<1h ago), `mathport` still had `is_R_or_C.abs`.

Reply:
@eric-wieser In fact, still waiting for leanprover-community/mathlib#18919